### ISSUE TITLE
conda clone fails in unexpected ways

### ISSUE THREAD

**Issue by ijstokes** [conda/conda#2458]
I get equivalent results with conda 4.0.5 and with 4.1.0rc1.  The results below are for 4.1.0rc1 ( @mcg1969 had suggested that this would do improved-but-different conda clone, so I decided to test it out here).

Summary of the situation:

* I need to clone the root environment (this is how we setup initial environments for projects)
* Stuff has been installed into the root environment which causes various errors
* The errors are surprising, because they are for packages that should be available both in the local package cache and from the channels that conda is configured to search
* One of the errors is for Python 2.7, which is always a big warning sign that something is badly mixed up (and if I knew what I'd tell you, but I haven't been able to figure it out)

```
$ sudo -u guest /opt/wakari/anaconda/bin/conda create -p /projects/guest/TestProject3/envs/de
fault --clone root
Fetching package metadata .............
src_prefix: '/opt/wakari/anaconda'
dst_prefix: '/projects/guest/TestProject3/envs/default'
Packages: 308
Files: 6
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/opt/wakari/anaconda/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 120, in main
    args_func(args, p)
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 127, in args_func
    args.func(args, p)
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/main_create.py", line 49, in execute
    install.install(args, parser, 'create')
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 225, in install
    clone(args.clone, prefix, json=args.json, quiet=args.quiet, index=index)
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 83, in clone
    quiet=quiet, index=index)
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/misc.py", line 246, in clone_env
    sorted_dists = r.dependency_sort(dists)
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/resolve.py", line 748, in dependency_sort
    depends = lookup(value)
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/resolve.py", line 743, in lookup
    return set(ms.name for ms in self.ms_depends(value + '.tar.bz2'))
  File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/resolve.py", line 577, in ms_depends
    rec = self.index[fkey]
KeyError: '<unknown>::pytables-3.2.2-np110py27_0.tar.bz2'
```

And here is the associated `conda info`:

```
$ sudo -u guest /opt/wakari/anaconda/bin/conda info 
Current conda install:

             platform : linux-64
        conda version : 4.1.0rc1
  conda-build version : 1.18.2
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /opt/wakari/anaconda  (read only)
  default environment : /opt/wakari/anaconda
     envs directories : /home/guest/.conda/envs
                        /home/guest/envs
                        /opt/wakari/anaconda/envs
        package cache : /home/guest/.conda/envs/.pkgs
                        /home/guest/envs/.pkgs
                        /opt/wakari/anaconda/pkgs
         channel URLs : https://conda.anaconda.org/r/linux-64/
                        https://conda.anaconda.org/r/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
                        https://conda.anaconda.org/wakari/linux-64/
                        https://conda.anaconda.org/wakari/noarch/
                        http://localhost:8080/conda/anaconda/linux-64/
                        http://localhost:8080/conda/anaconda/noarch/
          config file : /opt/wakari/anaconda/.condarc
    is foreign system : False
```

Since I thought that error was pretty strange (pytables is available in the listed linux-64 repo.continuum.io channel here: https://repo.continuum.io/pkgs/free/linux-64/), I thought I'd try to install it directly and see what happens:

```
$ sudo -u wakari /opt/wakari/anaconda/bin/conda install pytables
Fetching package metadata .............
Solving package specifications ....
Error: Package missing in current linux-64 channels: 
  - <unknown>::python 2.7*

You can search for this package on anaconda.org with

    anaconda search -t conda <unknown>::python 2.7*

You may need to install the anaconda-client command line client with

    conda install anaconda-client
```

And that seems pretty darn inscrutable.  Is it saying Python 2.7 is unavailable (because it isn't, it is in that same repo.continuum.io channel)?  And in any case, shouldn't conda complain because pytables is already installed?

```
$ sudo -u wakari /opt/wakari/anaconda/bin/conda list | grep pytables
pytables                  3.2.2               np110py27_0    <unknown>
```

So I am at a loss to figure out what is going on.  Advice urgently needed as to how to proceed (further debugging suggestions, identify what I've done wrong, or provide a work-around).  TIA!

**Comment by ijstokes** [comment]
This might be tied to #2459 but the initial error messages I see don't make it clear.

**Comment by mcg1969** [comment]
I’m pretty sure my fixes are not in RC1... I think RC2 is needed... but I could be wrong, and I am offsite so @kalefranz will have to help here

> On May 10, 2016, at 11:10 AM, Ian Stokes-Rees <notifications@github.com> wrote:
> 
> I get equivalent results with conda 4.0.5 and with 4.1.0rc1. The results below are for 4.1.0rc1 ( @mcg1969 <https://github.com/mcg1969> had suggested that this would do improved-but-different conda clone, so I decided to test it out here).
> 
> Summary of the situation:
> 
> I need to clone the root environment (this is how we setup initial environments for projects)
> Stuff has been installed into the root environment which causes various errors
> The errors are surprising, because they are for packages that should be available both in the local package cache and from the channels that conda is configured to search
> One of the errors is for Python 2.7, which is always a big warning sign that something is badly mixed up (and if I knew what I'd tell you, but I haven't been able to figure it out)
> $ sudo -u guest /opt/wakari/anaconda/bin/conda create -p /projects/guest/TestProject3/envs/de
> fault --clone root
> Fetching package metadata .............
> src_prefix: '/opt/wakari/anaconda'
> dst_prefix: '/projects/guest/TestProject3/envs/default'
> Packages: 308
> Files: 6
> An unexpected error has occurred, please consider sending the
> following traceback to the conda GitHub issue tracker at:
> 
>     https://github.com/conda/conda/issues
> 
> Include the output of the command 'conda info' in your report.
> 
> 
> Traceback (most recent call last):
>   File "/opt/wakari/anaconda/bin/conda", line 6, in <module>
>     sys.exit(conda.cli.main())
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 120, in main
>     args_func(args, p)
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 127, in args_func
>     args.func(args, p)
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/main_create.py", line 49, in execute
>     install.install(args, parser, 'create')
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 225, in install
>     clone(args.clone, prefix, json=args.json, quiet=args.quiet, index=index)
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 83, in clone
>     quiet=quiet, index=index)
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/misc.py", line 246, in clone_env
>     sorted_dists = r.dependency_sort(dists)
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/resolve.py", line 748, in dependency_sort
>     depends = lookup(value)
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/resolve.py", line 743, in lookup
>     return set(ms.name for ms in self.ms_depends(value + '.tar.bz2'))
>   File "/opt/wakari/anaconda/lib/python2.7/site-packages/conda/resolve.py", line 577, in ms_depends
>     rec = self.index[fkey]
> KeyError: '<unknown>::pytables-3.2.2-np110py27_0.tar.bz2'
> And here is the associated conda info:
> 
> $ sudo -u guest /opt/wakari/anaconda/bin/conda info 
> Current conda install:
> 
>              platform : linux-64
>         conda version : 4.1.0rc1
>   conda-build version : 1.18.2
>        python version : 2.7.11.final.0
>      requests version : 2.9.1
>      root environment : /opt/wakari/anaconda  (read only)
>   default environment : /opt/wakari/anaconda
>      envs directories : /home/guest/.conda/envs
>                         /home/guest/envs
>                         /opt/wakari/anaconda/envs
>         package cache : /home/guest/.conda/envs/.pkgs
>                         /home/guest/envs/.pkgs
>                         /opt/wakari/anaconda/pkgs
>          channel URLs : https://conda.anaconda.org/r/linux-64/
>                         https://conda.anaconda.org/r/noarch/
>                         https://repo.continuum.io/pkgs/free/linux-64/
>                         https://repo.continuum.io/pkgs/free/noarch/
>                         https://repo.continuum.io/pkgs/pro/linux-64/
>                         https://repo.continuum.io/pkgs/pro/noarch/
>                         https://conda.anaconda.org/wakari/linux-64/
>                         https://conda.anaconda.org/wakari/noarch/
>                         http://localhost:8080/conda/anaconda/linux-64/
>                         http://localhost:8080/conda/anaconda/noarch/
>           config file : /opt/wakari/anaconda/.condarc
>     is foreign system : False
> Since I thought that error was pretty strange (pytables is available in the listed linux-64 repo.continuum.io channel here: https://repo.continuum.io/pkgs/free/linux-64/ <https://repo.continuum.io/pkgs/free/linux-64/>), I thought I'd try to install it directly and see what happens:
> 
> $ sudo -u wakari /opt/wakari/anaconda/bin/conda install pytables
> Fetching package metadata .............
> Solving package specifications ....
> Error: Package missing in current linux-64 channels: 
>   - <unknown>::python 2.7*
> 
> You can search for this package on anaconda.org with
> 
>     anaconda search -t conda <unknown>::python 2.7*
> 
> You may need to install the anaconda-client command line client with
> 
>     conda install anaconda-client
> And that seems pretty darn inscrutable. Is it saying Python 2.7 is unavailable (because it isn't, it is in that same repo.continuum.io channel)? And in any case, shouldn't conda complain because pytables is already installed?
> 
> $ sudo -u wakari /opt/wakari/anaconda/bin/conda list | grep pytables
> pytables                  3.2.2               np110py27_0    <unknown>
> So I am at a loss to figure out what is going on. Advice urgently needed as to how to proceed (further debugging suggestions, identify what I've done wrong, or provide a work-around). TIA!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/conda/conda/issues/2458>

 
 
Michael C. Grant, Ph.D. <http://keybase.io/mcg>
Senior Solution Architect
Continuum Analytics, Inc. <http://continuum.io/>
Convex optimization resources:
CVX <http://cvxr.com/cvx/>, TFOCS <http://cvxr.com/tfocs/>, CVX Forum <http://ask.cvxr.com/>
CVX Research, Inc. <http://cvxr.com/>
 
  <http://continuum.io/>	  <http://cvxr.com/>

**Comment by kalefranz** [comment]
@ijstokes I believe this is fixed with 4.1.0rc2. Please reopen if you run into it again.
